### ISSUE TITLE
Inifinite resolution of Azure Function Host (v2) dependencies

### ISSUE THREAD

**Issue by Szer** [fsprojects/Paket#3172]
### Description

There is a [famous issue](https://github.com/Azure/azure-functions-host/issues/992) with binding redirects in Azure Functions. So I've tried to create a .lock file for Azure Function Host to help everyone check their packages against all the restrictions.
(Added: this is for runtime v2)

I took Host dependencices from [here](https://github.com/Azure/azure-functions-host/blob/master/src/WebJobs.Script/WebJobs.Script.csproj)
Binding redirects lies [here](https://github.com/Azure/azure-functions-host/blob/master/src/WebJobs.Script/app.config)

### Repro steps

1. Clone [this](https://github.com/Szer/azure-func-paket) repo

2. Paket Install

### Expected behavior

Appearance of paket.lock eventually

### Actual behavior

Infinite cycle of two resolutions

```
  Conflict detected:
   - Microsoft.CodeAnalysis.CSharp.Scripting 2.4.0 requested package Microsoft.CodeAnalysis.CSharp: 2.4
   - Microsoft.CodeAnalysis.CSharp.Workspaces 2.6.0 requested package Microsoft.CodeAnalysis.CSharp: 2.6
```

```
  Conflict detected:
   - NuGet.LibraryModel 4.4.0 requested package NuGet.Common: >= 4.4
   - NuGet.Configuration 4.6.2 requested package NuGet.Common: >= 4.6.2
```

### Known workarounds

None

**Comment by forki** [comment]
Yes, but isn't that exactly the point? Detecting conflicts in the
definition? I actually think it's correct. I noticed that azure functions
issue as well one day.

BTW we created the "external_lock" feature exactly for azure functions.
Unfortunately the official lock file from azure functions is only for
runtime v1

Szer <notifications@github.com> schrieb am Sa., 28. Apr. 2018, 09:20:

**Comment by Szer** [comment]
> BTW we created the "external_lock" feature exactly for azure functions.
Unfortunately the official lock file from azure functions is only for runtime v1

That's exactly what I want to do: create lock file for v2 :)

> Yes, but isn't that exactly the point? Detecting conflicts in the definition? I actually think it's correct. I noticed that azure functions issue as well one day.

But AFAIK azure-function-host is OK with these dependencies no matter the conflicts.
So you tell it is impossible to create lock file for v2 host?

**Comment by forki** [comment]
Well we need to understand the problem deeply and see if they really have a
conflict. Then we'd need to tell them. V2 is not released. Things can be
fixed.



Szer <notifications@github.com> schrieb am Sa., 28. Apr. 2018, 09:30:

**Comment by Szer** [comment]
> Then we'd need to tell them. V2 is not released. Things can be fixed.

They will, but I think it will take a long time to release it, but V2 already much better than V1 in terms of "dependency-hell" so a lot developers already uses it in production.

So I think it is a noble goal to provide a sane paket.lock file which will change accordingly with V2 host

**Comment by forki** [comment]
Yes I use it too. But not with external_lock

Szer <notifications@github.com> schrieb am Sa., 28. Apr. 2018, 09:49:
